src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -628,6 +628,19 @@ ErrorIfInsertSelectQueryNotSupported(Query *queryTree, RangeTblEntry *insertRte,
 	Query *subquery = NULL;
 	Oid selectPartitionColumnTableId = InvalidOid;
 
+	if (queryTree->resultRelation != 1)
+	{
```

Comment:
due to two observation,
resultRelation is expected to be 1, for insert queries. If it is not means postgresql intervened and added a new RTE to be used for insert. 
If a view is a target for insert, it first replaced by another view, then an actual relation rte.

I could not find a safer way of checking this. Open to suggestions.